IdentityCore/src/controllers/MSIDSilentController.m

```diff
@@ -114,7 +115,12 @@ - (void)acquireTokenWithRequest:(MSIDSilentTokenRequest *)request
     CONDITIONAL_START_EVENT(CONDITIONAL_SHARED_INSTANCE, self.requestParameters.telemetryRequestId, MSID_TELEMETRY_EVENT_API_EVENT);
     self.currentRequest = request;
     [request executeRequestWithCompletion:^(MSIDTokenResult *result, NSError *error)
-    {
+     {
+        MSIDAADRequestErrorHandler *handler = [[MSIDAADRequestErrorHandler alloc] init];
+        if (error && [handler shouldRetryNetworkingFailure:error.code]) {
+            completionBlock(result, error);
```

Comment:
This is missing a `return;` here, otherwise it will continue the regular flow and call `completionBlock` more than once.

Reply:
Very good catch!

Reply:
Added return statement, thank you!